src/c_kzg_4844.c

```diff
@@ -1089,12 +1089,12 @@ static C_KZG_RET hash_to_bytes(uint8_t out[32],
   if (bytes == NULL) return C_KZG_MALLOC;
 
   memcpy(bytes, FIAT_SHAMIR_PROTOCOL_DOMAIN, 16);
-  bytes_of_uint64(&bytes[16], n);
-  bytes_of_uint64(&bytes[16 + 8], FIELD_ELEMENTS_PER_BLOB);
+  bytes_of_uint64(&bytes[16], FIELD_ELEMENTS_PER_BLOB);
+  bytes_of_uint64(&bytes[16 + 8], n);
 
   for (i = 0; i < n; i++)
     for (j = 0; j < FIELD_ELEMENTS_PER_BLOB; j++)
-      bytes_from_bls_field(&bytes[ni + i * BYTES_PER_FIELD_ELEMENT], &polys[i][j]);
+      bytes_from_bls_field(&bytes[ni + BYTES_PER_FIELD_ELEMENT * (i * FIELD_ELEMENTS_PER_BLOB + j)], &polys[i][j]);
```

Comment:
From what I understand, this `bytes_from_bls_field` method copies 32 bytes into the `bytes` buffer at a time. So we are essentially flattening a 2d matrix into a 1d-array. The indexing was modifed to match this.

Reply:
@xrchz let me know if I misunderstood what this code is meant to be doing?

Reply:
this looks good to me - dunno what i was doing before :)